db/migrate/20230401123456_create_items.rb

```diff
@@ -0,0 +1,12 @@
+class CreateItems < ActiveRecord::Migration[7.0]
+  def change
+    create_table :items do |t|
+      t.string :name, null: false
+      t.decimal :price, precision: 10, scale: 2, null: false
+      t.decimal :discounted_price, precision: 10, scale: 2, null: false
+      t.references :collection, null: false, foreign_key: true
+
+      t.timestamps
+    end
+  end
+end
```

Comment:
The migration for creating items should include a check to ensure that 'discounted_price' is not greater than 'price'. This can be enforced at the database level with a check constraint.